lib/puppet/http/service/file_server.rb

```diff
@@ -2,25 +2,26 @@
 
 class Puppet::HTTP::Service::FileServer < Puppet::HTTP::Service
   API = '/puppet/v3'.freeze
-
-  EXCLUDED_FORMATS = [:yaml, :b64_zlib_yaml, :dot]
+  PATH_REGEX = /^\//
 
   def initialize(client, server, port)
     url = build_url(API, server || Puppet[:server], port || Puppet[:masterport])
     super(client, url)
   end
 
-  def get_file_metadata(mount_point:, path:, environment:, links: :manage, checksum_type: Puppet[:digest_algorithm], source_permissions: :ignore, ssl_context: nil)
-    headers = add_puppet_headers({ 'ACCEPT' => get_mime_types(Puppet::FileServing::Metadata).join(', ') })
+  def get_file_metadata(path:, environment:, links: :manage, checksum_type: Puppet[:digest_algorithm], source_permissions: :ignore, ssl_context: nil)
+    validate_path(path)
+
+    headers = add_puppet_headers({ 'Accept' => get_mime_types(Puppet::FileServing::Metadata).join(', ') })
 
     response = @client.get(
-      with_base_url("/file_metadata/#{mount_point}/#{path}"),
+      with_base_url("/file_metadata#{path}"),
       headers: headers,
       params: {
         links: links,
         checksum_type: checksum_type,
         source_permissions: source_permissions,
-        environment: environment
+        environment: environment.to_s
```

Comment:
This shouldn't be necessary. We should already be validating that environment is a valid type in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/http/client.rb#L198

Maybe this raise is being caught somewhere though

Reply:
Not sure why I added this, but I verified passing a non-string raises as it should: 

```
     Puppet::HTTP::SerializationError:
       HTTP REST queries cannot handle values of type 'Puppet::Node::Environment::Remote'
```

I'll take that commit out.